src/popover/popover.ts

```diff
@@ -150,14 +144,38 @@ export class NgbPopover implements OnInit, OnDestroy {
     });
   }
 
+  /**
+   * Content to be displayed as popover. If title and content are empty, the popover won't open.
+   */
+  @Input()
+  set ngbPopover(value: string | TemplateRef<any>) {
+    this._ngbPopover = value;
+    if (!this._ngbPopover && !this._popoverTitle && this._windowRef) {
+      this.close();
+    }
```

Comment:
@medinarkhov could you please extract this logic into a separate function? Even better, you could do this check in the `ngOnChanges` lifecycle hook - this way we wouldn't need a setter _and_ would remove code duplication.